icu4c/source/i18n/numrange_fluent.cpp

```diff
@@ -245,10 +245,10 @@ LocalizedNumberRangeFormatter::LocalizedNumberRangeFormatter(LocalizedNumberRang
         : LNF(static_cast<NFS<LNF>&&>(src)) {}
 
 LocalizedNumberRangeFormatter::LocalizedNumberRangeFormatter(NFS<LNF>&& src) noexcept
-        : NFS<LNF>(std::move(src)) {
+    : NFS<LNF>(std::move(src)) {
+    // Safely access the member from *this, which holds the moved-from state
     // Steal the compiled formatter
-    LNF&& _src = static_cast<LNF&&>(src);
-    auto* stolen = _src.fAtomicFormatter.exchange(nullptr);
+    auto* stolen = this->fAtomicFormatter.exchange(nullptr);
     delete fAtomicFormatter.exchange(stolen);
```

Comment:
Good catch on use-after-move, but... now we are stealing from ourselves, and putting the stolen pointer right back where it came from??
Do we need to futz with the fAtomicFormatter at all in this move copy constructor? If that got moved, then "this" should have the right state, right?
If we do need to futz with it, then I suspect we need to do it before moving stuff from src.

@aheninger @sffc please chime in. I don't have context for how these objects are built.

Reply:
I think that the pre-existing code is completely fine. `fAtomicFormatter`  is a field of LNF but not NFS. We read only that field, not any other fields.

Reply:
@sffc coverity complains that we are using `src` after it's been subject to `std::move`. This seems like a fair complaint.

Could we std::move just the NFS slice of src?
Could we move the formatter before moving src?

Reply:
I don't think it is a fair complaint. As far as I can tell, C++ does not offer a mechanism to mark only part of an object as having been moved (unlike Rust which provides such a mechanism). I found a thread that says that what this code is doing is the correct pattern when you have a child class calling the move constructor of a base class, although I don't know the credibility of the people responding on this thread:

https://cplusplus.com/forum/beginner/187808/